components/bl-drawer-component/src/helpers.js

```diff
@@ -0,0 +1,128 @@
+  [Placement.LEFT]  : 'translateX(-100%)',
+};
+
+export function useDrawerActions(component, eventHandlers, drawerContentRef, drawerContainerRef) {
+  const { onOpen, onClose } = eventHandlers;
+  const { placement, animationDuration, defaultOpen } = component;
+
+  const [visibility, setVisibility] = useState(false);
+
+  const openContent = event => {
+    event?.stopPropagation();
+
+    setVisibility(true);
+    onOpen();
+  };
+
+  const closeContent = useCallback(event => {
+    event?.stopPropagation();
+
+    if (!visibility) {
+      return;
+    }
+
+    drawerContentRef.current.style.transform = TransformMap[placement];
+    drawerContainerRef.current.classList.remove('backdrop');
+
+    setTimeout(() => setVisibility(false), animationDuration);
+    onClose();
+  }, [animationDuration, drawerContainerRef, drawerContentRef, onClose, placement, visibility]);
+
+  useEffect(() => {
+    if (defaultOpen) {
+      setVisibility(true);
+    }
+  }, [])
+
+  component.open = event => openContent(event);
+  component.close = event => closeContent(event);
+
+  return { visibility, openContent, closeContent };
+}
+
+export function useDrawerStyles(component) {
+  const { placement, size, backdropVisibility, animationDuration } = component;
+
+  const sizeStyles = useMemo(() => {
+    const validSize = ensureMeasure('width', size);
+
+    return SizeMap[placement](validSize);
+  }, [placement, size]);
+
+  const transitionDurationStyles = useMemo(() => (
+    { transitionDuration: animationDuration + 'ms' }
+  ), [animationDuration]);
+
+  const containerStyles = useMemo(() => (
+    backdropVisibility ? transitionDurationStyles : sizeStyles
+  ), [backdropVisibility, sizeStyles, transitionDurationStyles]);
+
+  const contentStyles = useMemo(() => {
+    const animationStyles = { ...transitionDurationStyles, transform: TransformMap[placement] };
+
+    return backdropVisibility ? { ...sizeStyles, ...animationStyles } : animationStyles;
+  }, [backdropVisibility, placement, sizeStyles, transitionDurationStyles]);
+
+  return { containerStyles, contentStyles };
+}
+
+export function useCloseEvents(drawerContentRef, closeContent, component) {
+  const { closeOnClickOutside, closeOnEscape } = component;
+
+  const handleEscClick = event => {
+    if (event.keyCode === ESCAPE_KEY_CODE) {
+      closeContent(event);
+    }
+  };
+
+  const handleClickOutsideDrawer = event => {
+    if (!drawerContentRef.current.contains(event.target)) {
+      closeContent(event);
+    }
+  };
+
+  useEffect(() => {
+    if (closeOnClickOutside) {
+      document.addEventListener('click', handleClickOutsideDrawer);
+    }
+
+    if (closeOnEscape) {
+      document.addEventListener('keydown', handleEscClick);
+    }
+
+    return () => {
+      document.removeEventListener('click', handleClickOutsideDrawer);
+      document.removeEventListener('keydown', handleEscClick);
+    };
+  }, []);
+}
+
+// TODO: Waiting for validation function from SDK Utils (MARKET-1835)
```

Comment:
Look at that, I have added this for the reason - WebStorm automatically highlights this mark(`TODO`)

There is nothing to change, just FYI.